src/AutoRest.CSharp/LowLevel/AutoRest/LowLevelTarget.cs

```diff
@@ -1,29 +1,73 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Generation.Types;
 using AutoRest.CSharp.Generation.Writers;
 using AutoRest.CSharp.Input;
 using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Output.Models;
 using AutoRest.CSharp.Output.Models.Types;
 
 namespace AutoRest.CSharp.AutoRest.Plugins
 {
     internal class LowLevelTarget
     {
-        public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel, SourceInputModel? sourceInputModel)
+        public static void Execute(GeneratedCodeWorkspace project, InputNamespace inputNamespace, SourceInputModel? sourceInputModel, bool cadlInput)
         {
-            var context = new BuildContext<LowLevelOutputLibrary>(codeModel, sourceInputModel);
-            foreach (var client in context.Library.RestClients)
+            var library = new DpgOutputLibraryBuilder(inputNamespace, sourceInputModel).Build(cadlInput);
+            foreach (var model in library.Models)
+            {
+                var codeWriter = new CodeWriter();
+                ModelWriter.WriteModel(codeWriter, model);
+                project.AddGeneratedFile($"{model.Type.Name}.cs", codeWriter.ToString());
+            }
+
+            foreach (var client in library.RestClients)
             {
                 var codeWriter = new CodeWriter();
                 var lowLevelClientWriter = new LowLevelClientWriter();
-                lowLevelClientWriter.WriteClient(codeWriter, client, context);
+                lowLevelClientWriter.WriteClient(codeWriter, client);
                 project.AddGeneratedFile($"{client.Type.Name}.cs", codeWriter.ToString());
             }
 
             var optionsWriter = new CodeWriter();
-            ClientOptionsWriter.WriteClientOptions(optionsWriter, context.Library.ClientOptions);
-            project.AddGeneratedFile($"{context.Library.ClientOptions.Type.Name}.cs", optionsWriter.ToString());
+            ClientOptionsWriter.WriteClientOptions(optionsWriter, library.ClientOptions);
+            project.AddGeneratedFile($"{library.ClientOptions.Type.Name}.cs", optionsWriter.ToString());
+        }
+
+        private class ModelWriter
+        {
+            public static void WriteModel(CodeWriter writer, ModelTypeProvider model)
```

Comment:
Not urgent. But I believe we need to add a `EnumTypeProvider` which accepts `InputEnumType`. And the `WriteModel` signature will be changed to `(CodeWriter writer, TypeProvider type)`.

Reply:
This can be done after merging.